packages/ts/react-crud/src/autogrid.tsx

```diff
@@ -156,21 +257,20 @@ function createDataProvider<TItem>(
       pageSize,
       sort,
     };
-
     const items = await service.list(req, filter.current);
-    let size;
-    if (items.length === pageSize) {
-      size = (pageNumber + 1) * pageSize + 1;
-
-      const cacheSize = (grid as GridElementWithInternalAPI<TItem>)._dataProviderController.rootCache.size;
-      if (cacheSize !== undefined && size < cacheSize) {
-        // Only allow size to grow here to avoid shrinking the size when scrolled down and sorting
-        size = undefined;
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    if (!(service as any).count) {
+      if (itemCountHolder.totalCount ?? itemCountHolder.filteredCount) {
+        console.error(
```

Comment:
This code runs with every single data provider request, so when scrolling it keeps logging errors. We should move it out of the data provider function.